docs/tutorials/migrating-your-ga4-data-to-snowplow/index.md

```diff
@@ -0,0 +1,204 @@
+```
+*Note this will error if you already have a custom role with this name*
+
+
+##### Grant the custom role to the service account
+```shell
+gcloud projects add-iam-policy-binding $PROJECT_ID --member serviceAccount:$SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com --role projects/$PROJECT_ID/roles/$ROLE_NAME
+```
+
+##### Create a key for the service account
+```shell
+gcloud iam service-accounts keys create $KEY_FILE_NAME --iam-account $SERVICE_ACCOUNT_NAME@$PROJECT_ID.iam.gserviceaccount.com
+```
+*This will download the key to your local machine in the same folder in which you are running the command*
+</TabItem>
+
+<TabItem value="Terraform" label="Terraform">
+##### Install gcloud
+See https://cloud.google.com/sdk/docs/install-sdk for installation instructions
+
+##### Authorize gcloud
+```shell
+gcloud auth application-default login
+```
+
+##### Install Terraform
+See https://learn.hashicorp.com/tutorials/terraform/install-cli for installation instructions
+
+##### Run Terraform
+There is a file in the root directory of the migrator tool called `manage_service_account.tf`. This file contains the Terraform code to create the service account.
+To create the service account in the root directory of the migrator tool, run the following commands:
+```shell
+terraform init
+terraform apply
+```
+The role ID is the name of the custom IAM role that will be created. The service account ID is the name of the service account that will be created.
+
+You can create a `terraform.tfvars` file if you do not want to re-enter the variables each time you run Terraform.
+E.g.
+```shell
+project_id = "project-123-abc"
+service_account_id = "ga4-migrator-service-account"
+role_id = "ga4_migrator_role"
+```
+The keyfile will be created in the same directory you ran Terraform.
+</TabItem>
+</Tabs>
+## Step 2 - How to run docker
+Set the environment variable SERVICE_ACCOUNT_KEY_FILE to the full path for the GCP service account key file.
+```shell
+export SERVICE_ACCOUNT_KEY_FILE=[path/to/service_account/keyfile.json]
+```
+
+If you are also going to run the export from Bigquery to Snowflake, set the following environment variables. This user must have ACCOUNTADMIN permissions in Snowflake.
+Also ensure that the GCS bucket you are using for the export already exists and is in the same location as your Bigquery table.
+
+```shell
+export SNOWFLAKE_USER=[snowflake_user]
+export SNOWFLAKE_PASSWORD=[snowflake_password]
+```
+If you do not already have Docker, installation instructions can be found (here)[https://docs.docker.com/get-docker/]  
+To run the docker container using docker compose. Run the following command -
+```shell
+docker compose up --build
+```
```

Comment:
yeah I can update the docs when we have the more final version of what it will look like. although the repo still has the tf file - that might be a bit hard for users to use if it's all in a docker container they pull?